lib/webdriveragent.js

```diff
@@ -302,18 +345,13 @@ class WebDriverAgent {
           return;
         }
         try {
-          let opts = {
-            method: 'GET',
-            uri: `${this.url.href}status`,
-            headers: 'Content-Type: application/json;charset=UTF-8, accept: application/json',
-            forever: true,
-            json: true,
-          };
-          let res = await request(opts);
-          if (res.status !== 0) {
-            throw new Error(`Received non-zero status code from WDA server: '${res.status}'`);
+          const proxyTimeout = this.noSessionProxy.timeout;
+          try {
+            this.noSessionProxy.timeout = this.wdaLaunchTimeout < 5000 ? this.wdaLaunchTimeout / 2 : 5000;
```

Comment:
yep, I'm not sure about the actual timeout value here. Probably, it would be smart to set the timeout to the lowest value possible, so it does not delay retries?